Makefile

```diff
@@ -54,14 +100,28 @@ $(FFMPEG_VERSION_PLATFORM):
 $(FFMPEG_BUILD): $(FFMPEG_VERSION_PLATFORM)
 	cd $(FFMPEG_VERSION_PLATFORM) && ./configure $(FFMPEG_OPTS) && $(MAKE) -j$(shell nproc) && $(MAKE) install
 
-build-ffmpeg:
-ifeq ($(UNAME_S),Linux)
-ifeq ($(UNAME_M),x86_64)
+build-ffmpeg: $(NDK_ROOT)
+# Only need nasm to build assembly kernels for x86 targets.
+ifeq ($(SOURCE_OS),linux)
+ifeq ($(SOURCE_ARCH),x86_64)
 	which nasm || (sudo apt update && sudo apt install -y nasm)
 endif
 endif
 	$(MAKE) $(FFMPEG_BUILD)
 
+# Download android NDK if it doesn't exist. Android NDK is only available
+# for linux x86_64 and darwin. On darwin, the NDK is installed via Android Studio.
+$(NDK_ROOT):
+ifeq ($(SOURCE_OS),linux)
+ifeq ($(SOURCE_ARCH),x86_64)
+ifeq ($(TARGET_OS),android)
+	sudo apt install -y unzip && \
+	wget https://dl.google.com/android/repository/android-ndk-r26-linux.zip && \
+	yes A | unzip android-ndk-r26-linux.zip
+endif
+endif
+endif
+
```

Comment:
Can you please either add support for automatically installing whatever Darwin needs to compile for android targets or providing instructions in the readme for how to install android studio (I personally don't know how to do that).